### PR TITLE
edge-23.7.3

### PR DESCRIPTION
**Pull request by adleong**
This edge release improves Linkerd's support for HttpRoute by allowing
`parent_ref` ports to be optional, allowing HttpRoutes to be defined in a
consumer's namespace, and adding support for the `ResponseHeaderModifier` filter.
It also fixes a panic in the destination controller.

* Added an option for disabling the network validator's security context for
  environments that provide their own
* Added high-availability (HA) mode for the multicluster service-mirror
* Added support for HttpRoute `parent_refs` that do not specify a port
* Fixed a Grafana error caused by an incorrect datasource (thanks @albundy83!)
* Added support for HttpRoutes defined in the consumer namespace
* Improved the granularity of logging levels in the control plane
* Fixed a race condition in the destination controller that could cause it to
  panic
* Added support for the `ResponseHeaderModifier` HttpRoute filter
* Updated extension CLI commands to prefer the `--register` flag over the
  `LINKERD_DOCKER_REGISTRY` environment variable, making the precedence more
  consistent (thanks @harsh020!)

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+
+This edge release improves Linkerd's support for HttpRoute by allowing
+`parent_ref` ports to be optional, allowing HttpRoutes to be defined in a
+consumer's namespace, and adding support for the ResposneHeaderModifier filter.
+It also fixes a panic in the destination controller.
+
+* Added an option for disabling the network validator's security context for
+  environments that provide their own
+* Added high-availability (HA) mode for the multicluster service-mirror
+* Added support for HttpRoute `parents_refs` that do not specify a port
```
Comment (hawkw) [inline CHANGES.md:13]: typo:
```suggestion
* Added support for HttpRoute `parent_refs` that do not specify a port
```

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+* Fixed a grafana error caused by an incorrect datasource (thanks @albundy83)
+* Added support for HttpRoutes defined in the consumer namespace
+* Improved the granularity of logging levels in the control plane
+* Fixed a race condition in the destination controller that could cause it to
+  panic
+* Added support for the ResponseHeaderModifier HttpRoute filter
```
Comment (hawkw) [inline CHANGES.md:19]: TIOLI: should this be 
```suggestion
* Added support for the `ResponseHeaderModifier` HttpRoute filter
```

File: CHANGES.md [hunk 3]
```diff
@@ -1,5 +1,23 @@
 # Changes
 
+## edge-23.7.3
+
+This edge release improves Linkerd's support for HttpRoute by allowing
+`parent_ref` ports to be optional, allowing HttpRoutes to be defined in a
+consumer's namespace, and adding support for the ResposneHeaderModifier filter.
```
Comment (hawkw) [inline CHANGES.md:7]: typo
```suggestion
consumer's namespace, and adding support for the ResponseHeaderModifier filter.
```

File: CHANGES.md [hunk 4]
```diff
@@ -1,5 +1,23 @@
...
+It also fixes a panic in the destination controller.
+
+* Added an option for disabling the network validator's security context for
+  environments that provide their own
+* Added high-availability (HA) mode for the multicluster service-mirror
+* Added support for HttpRoute `parents_refs` that do not specify a port
+* Fixed a grafana error caused by an incorrect datasource (thanks @albundy83)
```
Comment (hawkw) [inline CHANGES.md:14]:
```suggestion
* Fixed a Grafana error caused by an incorrect datasource (thanks @albundy83)
```

**Review by hawkw — COMMENTED** [review]
a few minor copyedits, but overall, LGTM!

File: charts/linkerd2-cni/Chart.yaml
Comment (alpeb) [inline charts/linkerd2-cni/Chart.yaml:1]: Nothing has changed in the linkerd2-cni chart, so there's no need to bump it (we need to update the instructions about this chart in RELEASE.md though).

**Review by alpeb — APPROVED** [review]

**Review by olix0r — APPROVED** [review]

**Pull request merged**